martin/src/srv/tiles_info.rs

```diff
@@ -58,6 +54,27 @@ async fn get_source_info(
     Ok(HttpResponse::Ok().json(merge_tilejson(&sources, tiles_url)))
 }
 
+fn generate_tiles_base(
```

Comment:
* if `srv_config.base_path` is set, you should not parse it on every request.  It should be in the right format, parsed during config parsing.
* I would inline `generate_tiles_base` here -- the whole `get_source_info` is about constructing `tiles_url` - it would be confusing to break it into two functions I think, and creates unneeded complexities with error handling.
* `generate_tiles_base` should not create Strings (clones), but instead `tiles_path` should be a `&str` that will be used in the `let path_and_query = ...`